### ISSUE TITLE
Feature: drawing and quoting

### ISSUE THREAD

**Issue by quozl** [sugarlabs/chat#17]
Some of the original design features not yet implemented are drawing and quoting.

* [ ] increase the height of the entry widget and accept touchscreen or mouse drawing, as well as text,
* [ ] transmit the drawing and text to the shared activity instance,
* [ ] receive and render both the drawing and the text,
* [ ] add a clear entry button to the entry widget,
* [ ] in the chat history, for each transmission add a _quote_ button, which will load the entry widget with the drawing and text.

Reference: http://wiki.laptop.org/go/Chat

**Comment by Himanshu-Vishwas** [comment]
Hi! I would like to contribute to the issue.

**Comment by quozl** [comment]
That's wonderful, but it has nothing to do with the issue, so I'll hide your comment and my reply as off-topic.  Please only discuss the issue.

**Comment by RuiChen-03** [comment]
hi, for the fourth issue, may I ask what 'the enrty widget' refer to ? Is it the enrty for write message box or anything alse?

**Comment by chimosky** [comment]
> hi, for the fourth issue, may I ask what 'the enrty widget' refer to ? Is it the enrty for write message box or anything alse?

Yes, the entry widget being referred to in the opening comment is the one for writing messages.

**Comment by RuiChen-03** [comment]
> > hi, for the fourth issue, may I ask what 'the enrty widget' refer to ? Is it the enrty for write message box or anything alse?
> 
> Yes, the entry widget being referred to in the opening comment is the one for writing messages.

thank you, and I have tried to realize the fourth task and submitted the PR already,.hope it can work

**Comment by chimosky** [comment]
Updating the task list; 

* [ ]  increase the height of the entry widget and accept touchscreen or mouse drawing, as well as text,
* [ ]  transmit the drawing and text to the shared activity instance,
* [ ]  receive and render both the drawing and the text,
* [x]  add a clear entry button to the entry widget,
* [ ]  in the chat history, for each transmission add a _quote_ button, which will load the entry widget with the drawing and text.

**Comment by Ovalelephant35** [comment]
Task number 5 , chat history and quote button what we are aiming here to do, like chat which is already written there is current session its drawing and text?

**Comment by quozl** [comment]
Sorry, I can't parse your question.  Although it was designed many years ago by our volunteers, I think this feature can be found in many other instances of chat software today.  I note the page doesn't load.  I'll add that to my list.

**Comment by Ovalelephant35** [comment]
Ok, I thinking it is what I am thinking, respond to or elaborate on the selected message without having to retype or redraw the content.

**Comment by quozl** [comment]
No, I was not proposing message selection.  When the new message is sent, I would expect it to contain the drawing and text with any changes made by the user.
